resources/assets/js/store/modules/permissions.js

```diff
@@ -0,0 +1,162 @@
+import api from 'plugins/http/api';
+
+/**
+ * Simple interface for interacting with user roles and permissions
+ * 
+ * permissions is an array with the following structure
+ * 
+[
+		{
+			"name": "Create Subsites",
+			"slug": "subsite.create",
+			"roles": [
+				"site.owner"
+			]
+		},
+		...
+ */
+const state = {
```

Comment:
Would be nice to have a docblock describing the datatype / structure of these variables especially as you are storing the permissions (+roles) in the roles variable (I think?)